lib/src/models/scoped_function_declaration.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:meta/meta.dart';
+
+import 'function_type.dart';
+import 'scoped_class_declaration.dart';
+
+/// Represents a declaration of function / method.
+@immutable
+class ScopedFunctionDeclaration {
+  /// Type of declared function entity.
+  final FunctionType type;
+
+  /// The node that represents in the AST structure for a Dart program.
+  final Declaration declaration;
+
+  /// The class declaration of the class to which this function belongs.
+  final ScopedClassDeclaration enclosingDeclaration;
+
+  /// Returns user defined name.
+  String get name {
+    if (declaration is FunctionDeclaration) {
+      return (declaration as FunctionDeclaration).name?.name;
+    } else if (declaration is ConstructorDeclaration) {
+      return (declaration as ConstructorDeclaration).name?.name ??
+          (declaration.parent as NamedCompilationUnitMember).name?.name;
+    } else if (declaration is MethodDeclaration) {
+      return (declaration as MethodDeclaration).name?.name;
+    }
+
+    return null;
```

Comment:
Why return null here and empty string [here](https://github.com/dart-code-checker/dart-code-metrics/pull/230/files#diff-d7f66dd0ebfb0a270e96a557740e36f9909320394ff085a6520d10cff38b1d97R23)?